src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1387,30 +1384,25 @@ AddUnionAllSetOperationsToAttributeEquivalenceClass(AttributeEquivalenceClass **
 }
 
 
+#if PG_VERSION_NUM >= PG_VERSION_13
 /*
- * RangeTableOffsetCompat returns the range table offset(in glob->finalrtable) for the appendRelInfo.
- * For PG < 13 this is a no op.
+ * FindPreviousParentAmount returns the number of parents that come before
+ * the parent of the given append rel info.
  */
-static int
-RangeTableOffsetCompat(PlannerInfo *root, AppendRelInfo *appendRelInfo)
-{
-	#if PG_VERSION_NUM >= PG_VERSION_13
-	int i = 1;
-	for (; i < root->simple_rel_array_size; i++)
-	{
-		RangeTblEntry *rte = root->simple_rte_array[i];
-		if (rte->inh)
-		{
-			break;
+static int FindPreviousParentAmount(List* appendRelList, AppendRelInfo* targetAppendRelInfo) {
```

Comment:
Maybe call this: `ParentCountPriorToAppendRel()`  and do `curParentIndex <= targetParentIndex` in the code. And, then:
```C
int parentCount = ParentCountPriorToAppendRel(root->append_rel_list, appendRelInfo);
int skipParentCount = parentCount -1;
```
That looks more natural and easier to follow to me. What do you think?

Reply:
yes that is easier to follow